api/api-member/src/main/java/com/pgms/apimember/service/MemberService.java

```diff
@@ -0,0 +1,100 @@
+import com.pgms.apimember.dto.response.MemberDetailGetResponse;
+import com.pgms.apimember.exception.CustomErrorCode;
+import com.pgms.apimember.exception.MemberException;
+import com.pgms.coredomain.domain.member.Member;
+import com.pgms.coredomain.domain.member.repository.MemberRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class MemberService {
+
+	private final MemberRepository memberRepository;
+
+	@Transactional(readOnly = true)
+	public MemberDetailGetResponse getMemberDetail(Long memberId) {
+		return MemberDetailGetResponse.from(
+			memberRepository.findByIdWithRole(memberId).
+				orElseThrow(() -> new MemberException(CustomErrorCode.MEMBER_NOT_FOUND)));
+	}
+
+	@Transactional(readOnly = true)
+	public void verifyPassword(Long memberId, String password) {
+		final Member member = getAvailableMember(memberId);
+		validatePlainPassword(password, member.getPassword());
+	}
+
+	public void updateMember(Long memberId, MemberInfoUpdateRequest requestDto) {
+		final Member member = getAvailableMember(memberId);
+		member.updateMemberInfo(
+			requestDto.name(),
+			requestDto.phoneNumber(),
+			requestDto.birthDate(),
+			requestDto.gender(),
+			requestDto.streetAddress(),
+			requestDto.detailAddress(),
+			requestDto.zipCode()
+		);
+	}
+
+	public void updatePassword(Long memberId, MemberPasswordUpdateRequest requestDto) {
+		final Member member = getAvailableMember(memberId);
+		validatePlainPassword(requestDto.originPassword(), member.getPassword());
+		validatePasswordAndConfirm(requestDto.newPassword(), requestDto.newPasswordConfirm());
+		member.updatePassword(requestDto.newPassword()); // TODO: 비밀번호 암호화 추가 필요
+
+	}
+
+	public void deleteMember(Long memberId) {
+		final Member member = getAvailableMember(memberId);
+		member.updateToDeleted();
+	}
+
+	public Long restoreMember(Long memberId) {
+		final Member member = memberRepository.findById(memberId)
+			.orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND));
+
+		if (member.isDeleted()) {
+			member.updateToActive();
+		}
+
+		return member.getId();
+	}
+
+	private void validatePlainPassword(String plainPassword, String encodedPassword) {
+		// TODO: 비밀번호 암호화 추가 필요
+		// encoder.matches(plainPassword, encodedPassword);
+		if (!plainPassword.equals(encodedPassword)) { // 암호화 없는 임시 검증
+			throw new MemberException(PASSWORD_NOT_MATCHED);
+		}
+	}
+
+	private void validatePasswordAndConfirm(String password, String passwordConfirm) {
+		if (!password.equals(passwordConfirm)) {
+			throw new MemberException(PASSWORD_CONFIRM_NOT_MATCHED);
+		}
+	}
+
+	private Member getAvailableMember(Long memberId) {
+		final Member member = memberRepository.findById(memberId)
+			.orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND));
+
+		// 필터단에서 멤버 삭제여부 검증 시 필요하지 않을수도 있음
+		if (member.isDeleted()) {
+			throw new MemberException(MEMBER_ALREADY_DELETED);
+		}
+		return member;
```

Comment:
```suggestion
      return memberRepository.findByIdAndIsDeletedFalse(memberId)
          .orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND));
```
개인적인 의견입니다만 한줄로 작성할 수도 있을거 같아요..? 
available한 회원을 검증하기 위한 로직으로 보이는데요, 주석대로 필터에서 삭제 여부를 검증할 수도 있겠네요!